radish34/__tests__/integration.test.js

```diff
@@ -158,12 +220,30 @@ describe('Buyer sends new RFP to supplier', () => {
       expect(res.body.data.rfp.sku).toEqual(sku);
     });
 
-    test('Supplier graphql query rfp() returns 200', async () => {
+    test('Supplier1 graphql query rfp() returns 200', async () => {
+      const queryBody = `{ rfp(uuid: "${rfpId}") { _id, sku } } `
+      // Wait for db to update
+      let res;
+      for (let retry = 0; retry < 10; retry++) {
+        res = await request(supplier1ApiURL)
+          .post('/graphql')
+          .send({ query: queryBody });
+        if (res.body.data.rfp !== null) {
+          break;
+        }
+        await new Promise((r) => setTimeout(r, 1000));
+      }
+      expect(res.statusCode).toEqual(200);
+      expect(res.body.data.rfp._id).toEqual(rfpId);
+      expect(res.body.data.rfp.sku).toEqual(sku);
+    });
+
+    test('Supplier2 graphql query rfp() returns 200', async () => {
       const queryBody = `{ rfp(uuid: "${rfpId}") { _id, sku } } `
       // Wait for db to update
       let res;
       for (let retry = 0; retry < 10; retry++) {
```

Comment:
Is there a way these retries are made based on a parameter, so that user would know which test in the overall process is the wait/delay at?

Reply:
Are you asking if we can print logs while the test is running so the user knows which test is currently being run? Or are you suggesting that the number of retries is configurable and can be passed in by the user as a parameter?

Reply:
i mean logs for a specific function so that user knows which test is currently the testing process is at

Reply:
Added some logs to long-running tests so user is more informed